eclipse-glsp/glsp-server
# Update target platform to r2024-09

#### What it does

- Update target platforms.
   -New default target Eclipse 2024-09
  - New minimum baseline Eclipse 2023-12
  - Switch to jarkarta variant of lsp4j websocket
  - Switch to Jetty 12

Builds on top of: https://github.com/eclipse-glsp/glsp-server/pull/24

Changes done on top:
- Remove outdated targets
- Adjust version ranges
- Create a new target definition for the minimum supported version (2023-12)
- Restore explicit reexport of certain packages to maintain backwards compatibility
- Fix and align m2 build
- Add possibility to change the target platform of the p2 build with the p2.target property

Part of https://github.com/eclipse-glsp/glsp/issues/1397

#### How to test


Building and running the example via CLI should work as expected
Test the following:
P2:
- mvn clean verify -Pp2
- mvn clean  verifiy -Dp2.target=r2023-12 (to build against the minimum p2 target)
M2:
- mvn clean verify -Pm2
Build the example fatjar
- mvn clean verify -Pm2 -Pfatjar
Start the example jar
- java -jar examples/org.eclipse.glsp.example.workflow/target/org.eclipse.glsp.example.workflow-2.3.0-SNAPSHOT-glsp.jar -p 80
81 --websocket
(and open the example diagram.html of the standalone client example)

Eclipse version used for development: 2024-06 Modelling Tools
Also tested with 2023-12 Modellingtools

#### Follow-ups



#### Changelog
